app/src/main/java/com/example/sharingang/NewEditFragment.kt

```diff
@@ -161,10 +174,45 @@ class NewItemFragment : Fragment() {
     }
 
     private fun updateLocation(location: Location) {
-        binding.latitude = location.latitude.toString()
-        binding.longitude = location.longitude.toString()
+        updateLocationWithCoordinates(location.latitude, location.longitude)
+    }
+
+    private fun updateLocationWithCoordinates(latitude: Double, longitude: Double) {
+        binding.latitude = latitude.toString()
+        binding.longitude = longitude.toString()
         val address =
-            geocoder.getFromLocation(location.latitude, location.longitude, 1).getOrNull(0)
+            geocoder.getFromLocation(latitude, longitude, 1).getOrNull(0)
         binding.postalAddress.text = address?.getAddressLine(0) ?: ""
     }
+
+    private fun setupNewOrEditFragment() {
+        if (viewModel.onEdit.value == false) {
```

Comment:
Can we do `existingItem = args.item` before and check if it's null to determine if we are creating a new item or editing an existing item?

Reply:
Corrected, thanks!